dev/breeze/src/airflow_breeze/ci/build_image.py

```diff
@@ -0,0 +1,135 @@
+#         if len(param_value) > 0:
+#             args_command+= " --build-arg " + verify_param.upper() + "=" + param_value
+#     args_command += " " +ci_image.docker_cache_ci_directive
+#     return args_command
+# My way
+def construct_arguments_docker_command(ci_image: BuildParams) -> List[str]:
+    args_command = []
+    for param in PARAMS_CI_IMAGE:
+        args_command.append("--build-arg")
+        args_command.append(param.upper() + "=" + str(getattr(ci_image, param)))
+    for verify_param in PARAMS_TO_VERIFY_CI_IMAGE:
+        param_value = str(getattr(ci_image, verify_param))
+        if len(param_value) > 0:
+            args_command.append("--build-arg")
+            args_command.append(verify_param.upper() + "=" + param_value)
+    docker_cache = ci_image.docker_cache_ci_directive
+    if len(docker_cache) > 0:
+        args_command.extend(ci_image.docker_cache_ci_directive)
+    return args_command
+
+
+# Suggested way
+# def construct_arguments_docker_command(ci_image: BuildParams) -> List[Any]:
+#     args_command = []
+#     for param in PARAMS_CI_IMAGE:
+#         args_command.append([" --build-arg " + param.upper() + "=" + str(getattr(ci_image, param))])
+#     for verify_param in PARAMS_TO_VERIFY_CI_IMAGE:
+#         param_value = str(getattr(ci_image, verify_param))
+#         if len(param_value) > 0:
+#             args_command.append([" --build-arg " + verify_param.upper() + "=" + param_value])
+#     args_command.append(ci_image.docker_cache_ci_directive)
+#     return args_command
+# New way
+def construct_docker_command(ci_image: BuildParams) -> List[str]:
+    arguments = construct_arguments_docker_command(ci_image)
+    console.print('Arguments:', arguments)
+    final_command = []
+    final_command.extend(["docker", "build"])
+    final_command.extend(arguments)
+    final_command.extend(["-t", ci_image.airflow_ci_image_name, "--target", "main", "."])
+    final_command.extend(["-f", str(Path(get_airflow_sources_root(), 'Dockerfile.ci').resolve())])
+    return final_command
+
+
+# Old way
+# def construct_docker_command(ci_image: BuildParams):
+#     arguments = construct_arguments_docker_command(ci_image)
+#     console.print('Arguments:', arguments)
+#     final_command = (
+#         "docker build"
+#         + arguments
+#         + f" -t {ci_image.airflow_ci_image_name} --target main . -f "
+#         + str(Path(get_airflow_sources_root(), 'Dockerfile.ci').resolve())
+#     )
+#     return final_command
+
+
+def build_image(verbose, **kwargs):
+    ci_image_params = BuildParams(filter_out_none(**kwargs))
+    is_cached, value = check_cache_and_write_if_not_cached(
+        "PYTHON_MAJOR_MINOR_VERSION", ci_image_params.python_version
+    )
+    if is_cached:
+        ci_image_params.python_version = value
+    cmd = construct_docker_command(ci_image_params)
+    # console.print("Command to run:", cmd, style="info")
+    output = run_command(cmd, capture_output=True, text=True)
```

Comment:
One small change here. We should just not capture the output (yet). This will help to see progress while the command is running.

Reply:
Oh i see. i will remove capture_output and see if i could see the progress and update here.